src/Utilities/Compiler/Options/AnalyzerOptionsExtensions.cs

```diff
@@ -644,38 +603,14 @@ SymbolNamesWithValueOption<TValue> GetDefaultValue()
                 return categorizedAnalyzerConfigOptions;
             }
 
-            var createValueCallback = new ConditionalWeakTable<AnalyzerOptions, ICategorizedAnalyzerConfigOptions>.CreateValueCallback(_ => ComputeCategorizedAnalyzerConfigOptions());
-            return s_cachedOptions.GetValue(options, createValueCallback);
-
-            // Local functions.
-            ICategorizedAnalyzerConfigOptions ComputeCategorizedAnalyzerConfigOptions()
-            {
-                var categorizedOptionsFromAdditionalFiles = ComputeCategorizedAnalyzerConfigOptionsFromAdditionalFiles();
-
+            var createValueCallback = new ConditionalWeakTable<AnalyzerOptions, ICategorizedAnalyzerConfigOptions>.CreateValueCallback(_ =>
 #if CODEANALYSIS_V3_OR_BETTER
```

Comment:
Projects where this is not true:

- PublicAPIAnalyzers
- PerformanceSensitiveAnalyzers
- BannedAPIAnalyzers

Need to give a closer look to these since I believe there will be behavioral change if they come to this code path.

If they don't come to this code path, it makes more sense to move this conditional directive further up.